src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/visit/BookVisitTest.kt

```diff
@@ -647,10 +682,18 @@ class BookVisitTest : IntegrationTestBase() {
       .jsonPath("$.developerMessage").isEqualTo("Visit with booking reference - $reference is in the past, it cannot be changed")
   }
 
-  private fun assertAuditEvent(visitDto: VisitDto, visitEntity: Visit) {
+  private fun assertAuditEvent(visitDto: VisitDto, visitEntity: Visit, userType: UserType = STAFF) {
     val eventAudit = this.eventAuditRepository.findLastEventByBookingReference(visitDto.reference)
     assertThat(eventAudit.type).isEqualTo(EventAuditType.BOOKED_VISIT)
-    assertThat(eventAudit.actionedBy).isEqualTo("booking_guy")
+    assertThat(eventAudit.actionedBy).isNotNull()
+    assertThat(eventAudit.actionedBy.userType).isEqualTo(userType)
+    if (STAFF == eventAudit.actionedBy.userType) {
+      assertThat(eventAudit.actionedBy.userName).isEqualTo("booking_guy")
```

Comment:
could assert that the other field is null ? To make sure we're only inserting into the 1 column per user type?